### PR TITLE
Link subscription vpn android

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/25866
Resolves https://github.com/brave/brave-browser/issues/21972
Resolves https://github.com/brave/brave-browser/issues/26355

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Coming soon

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/settings/BraveVpnPreferences.java [hunk 1]
```diff
@@ -159,6 +163,19 @@ public boolean onPreferenceClick(Preference preference) {
                         return true;
                     }
                 });
+        findPreference(PREF_LINK_SUBSCRIPTION)
+                .setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
+                    @Override
+                    public boolean onPreferenceClick(Preference preference) {
+                        // BraveVpnUtils.getBraveAccountUrl();
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/settings/BraveVpnPreferences.java:170]: nit: seems like can be removed

**Review by Pavneet-Sing — APPROVED** [review]

**Comment by bsclifton** [comment]
@deeppandya needs a rebase - also we'll want to create a new feature for both the notification AND the button which we can "griffinize". See https://github.com/brave/brave-browser/issues/26355

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java [hunk 2]
```diff
@@ -957,6 +958,10 @@ public void finishNativeInitialization() {
                         BraveVpnUtils.SUBSCRIPTION_PARAM_TEXT, getPackageName());
             }
         }
+        if (BraveVpnPrefUtils.isSubscriptionPurchase()
+                && !BraveVpnPrefUtils.isLinkSubscriptionDialogShown()) {
+            showLinkVpnSubscriptionDialog();
```
Comment (bsclifton) [inline android/java/org/chromium/chrome/browser/app/BraveActivity.java:963]: Can this also check a feature flag? So we can gate the showing of this dialog on enabling via Griffin?

Also curious - do you have a screen cap of the dialog itself? (maybe you can update PR to include pic)

**Review by bsclifton — APPROVED** [review]
Feature flag changes look great! 🙂👍

**Review by mkarolin — APPROVED** [review]
`chromium_src` change LGTM, but please see the new way of declaring/defining features with macros.

**Comment by srirambv** [comment]
Verification passed on Oppo Reno 5 with Android 12 running 1.47.85 x64 Nightly build
- Verified steps from PR 
- Verified VPN settings has a new item for linking subscription
- Verified clicking the setting opens a new tab and loads `https://account.brave.com`
- Verified link subscription modal is shown on second restart
- Verified `Link purchase to your Brave account` is only enabled when there is an active subscription on Android IAP
- Verified when the setting is disabled the modal is also not shown irrespective how many times the browser is restarted

Link Subscription setting | Link Subscription Modal (Light) | Link Subscription Modal (Dark)
-- | -- | --
image | image | image

**Pull request merged**